sdc/src/util/json.d

```diff
@@ -805,27 +812,27 @@ public:
                 return false;
             }
 
-            final switch (_type) with(JsonType) {
-		        case Bool:
-		            return _boolean == other._boolean;
-		        case Int:
-		            return _integer == other._integer;
-		        case Float:
-		            return _floating == other._floating;
-		        case String:
-		            return _str == other._str;
-		        case Array:
-		            return _array == other._array;
-		        case Object:
-		            return _object == other._object;
-		        case Null:
-		            // The types match, so this is true.
-		            return true;
+            with(JSON_TYPE) final switch (_type) {
+            case BOOL:
+                return _boolean == other._boolean;
+            case INT:
+                return _integer == other._integer;
+            case FLOAT:
+                return _floating == other._floating;
+            case STRING:
+                return _str == other._str;
+            case ARRAY:
+                return _array == other._array;
+            case OBJECT:
+                return _object == other._object;
+            case NULL:
+                // The types match, so this is true.
+                return true;
             }
-        } else static if(is(T : string)) {
-            return _type == JsonType.String && _str == other;
+        } else static if(is(T : const(char[]))) {
```

Comment:
didn't catch `opEquals!string`, as in https://github.com/SDC-Developers/SDC/pull/188/files#diff-80c2c9a6755b5f6f668dd80fb157e052R2640